homeassistant/components/onvif/camera.py

```diff
@@ -119,29 +116,9 @@ async def async_camera_image(self):
         image = None
 
         if self.device.capabilities.snapshot:
-            auth = None
-            if self.device.username and self.device.password:
-                if self._basic_auth:
-                    auth = HTTPBasicAuth(self.device.username, self.device.password)
-                else:
-                    auth = HTTPDigestAuth(self.device.username, self.device.password)
-
-            def fetch():
-                """Read image from a URL."""
-                try:
-                    response = requests.get(self._snapshot_uri, timeout=5, auth=auth)
-                    if response.status_code < 300:
-                        return response.content
-                except requests.exceptions.RequestException as error:
-                    LOGGER.error(
-                        "Fetch snapshot image failed from %s, falling back to FFmpeg; %s",
-                        self.device.name,
-                        error,
-                    )
-
-                return None
-
-            image = await self.hass.async_add_executor_job(fetch)
+            image = await self.device.device.get_snapshot(
+                self.profile.token, self._basic_auth
+            )
```

Comment:
```suggestion
            try:
                image = await self.device.device.get_snapshot(
                    self.profile.token, self._basic_auth
                )
            except ONVIFError as error:
                LOGGER.error(
                    "Fetch snapshot image failed from %s, falling back to FFmpeg; %s",
                    self.device.name,
                    error,
                )
```

Reply:
I would also replace:
```python
async def async_get_snapshot_uri(self, profile: Profile) -> str:
    """Get the snapshot URI for a specified profile."""
    if not self.capabilities.snapshot:
        return None

    media_service = self.device.create_media_service()
    req = media_service.create_type("GetSnapshotUri")
    req.ProfileToken = profile.token
    result = await media_service.GetSnapshotUri(req)
    return result.Uri
```
in `device.py` with:
```python
async def async_get_snapshot(self, profile, basic_auth=False):
    """Get a snapshot from the given url."""
    return await self.device.get_snapshot(profile.token, basic_auth)
```
and use it here and in `__init__.py`

Reply:
There's no need to use it here or in init, because the library now wraps and caches that for us.  I will delete it from device.py since it's not used anymore though.